tests/js/client/server_parameters/test-export-shard-usage-metrics-bytes-per-shard-cluster.js

```diff
@@ -0,0 +1,1744 @@
+          let parsed = getParsedMetrics(db._name(), cn);
+          assertFalse(parsed.hasOwnProperty("reads"), parsed);
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+    
+    testHasMetricsWhenWritingIntoCollectionBatchInsert : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          const n = 25;
+          let docs = [];
+          for (let i = 0; i < n; ++i) {
+            docs.push({ value: i });
+          }
+          c.insert(docs);
+          
+          let parsed = getParsedMetrics(db._name(), cn);
+          assertFalse(parsed.hasOwnProperty("reads"), parsed);
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+    
+    testHasMetricsWhenWritingIntoCollectionSingleRemoves : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          // must insert first to remove something later
+          const n = 50;
+          let docs = [];
+          for (let i = 0; i < n; ++i) {
+            docs.push({ _key: "test" + i, value: i });
+          }
+          c.insert(docs);
+
+          for (let i = 0; i < n; ++i) {
+            c.remove("test" + i);
+          }
+          
+          let parsed = getParsedMetrics(db._name(), cn);
+          assertFalse(parsed.hasOwnProperty("reads"), parsed);
+          // count 40-50 bytes for each insert, and 10-20 bytes for each remove
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor + n * 10 * replicationFactor, {parsed, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor + n * 20 * replicationFactor, {parsed, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+    
+    testHasMetricsWhenWritingIntoCollectionBatchRemove : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          // must insert first to remove something later
+          const n = 50;
+          let docs = [];
```

Comment:
This code is repeated all over the place. We could factor this out in a function.